cf-custom-resources/lib/alb-rule-priority-generator.js

```diff
@@ -158,22 +158,41 @@ const getListenerRules = async function (listenerArn) {
  */
 exports.nextAvailableRulePriorityHandler = async function (event, context) {
   let responseData = {};
+  let nextRootRuleNumber, nextNonRootRuleNumber;
   const physicalResourceId =
-    event.PhysicalResourceId || `alb-rule-priority-${event.LogicalResourceId}`;
-  let isRootPath = event.ResourceProperties.RulePath === "/";
-
+      event.PhysicalResourceId || `alb-rule-priority-${event.LogicalResourceId}`;
   try {
     switch (event.RequestType) {
       case "Create":
       case "Update":
-        if (isRootPath) {
-          responseData.Priority = await calculateNextRootRulePriority(
-            event.ResourceProperties.ListenerArn
-          );
+        if (event.ResourceProperties.RulePath.length == 1){
+          if (event.ResourceProperties.RulePath[0] === "/") {
+            responseData.Priority = await calculateNextRootRulePriority(
+                event.ResourceProperties.ListenerArn
+            );
+          } else {
+            responseData.Priority = await calculateNextRulePriority(
+                event.ResourceProperties.ListenerArn
+            );
+          }
         } else {
-          responseData.Priority = await calculateNextRulePriority(
-            event.ResourceProperties.ListenerArn
-          );
+          for (let i=0; i < event.ResourceProperties.RulePath.length; i++){
```

Comment:
Is it typical to increment/decrement by 1? If more listener rules are added later, is it possible for the priority to be between existing ones?

Reply:
Good question - if you look at this existing [test case](https://github.com/aws/copilot-cli/blob/5c0124b09eeb99e7d75b8fa6032b06653ddbe57c/cf-custom-resources/test/alb-rule-priority-generator-test.js#L232) which has priority 3 and 5 assigned. But whenever new listener rule comes in, it assigns priority as 6 and not 4 which is between 3-5 and also available. I think we do it because additional rules are for more specific route paths which should get the most highest priority hence we do what we are doing here.